examples/lib/stories/collision_detection/quadtree_example.dart

```diff
@@ -0,0 +1,434 @@
+    }
+
+    staticLayer.reRender();
+    camera.viewport = FixedResolutionViewport(Vector2(500, 250));
+    final playerPoint = Vector2.all(mapSize * tileSize / 2);
+
+    final player =
+        Player(position: playerPoint, size: Vector2.all(tileSize), priority: 2);
+    add(player);
+    this.player = player;
+    camera.followComponent(player);
+
+    final brick = Brick(
+      position: playerPoint.translate(0, -tileSize * 2),
+      size: Vector2.all(tileSize),
+      priority: 0,
+      sprite: spriteBrick,
+    );
+    add(brick);
+    staticLayer.components.add(brick);
+
+    final water1 = Water(
+      position: playerPoint.translate(0, tileSize * 2),
+      size: Vector2.all(tileSize),
+      priority: 0,
+      sprite: spriteWater,
+    );
+    add(water1);
+
+    final water2 = Water(
+      position: playerPoint.translate(tileSize * 2, 0),
+      size: Vector2.all(tileSize),
+      priority: 0,
+      sprite: spriteWater,
+    );
+    add(water2);
+
+    final water3 = Water(
+      position: playerPoint.translate(-tileSize * 2, 0),
+      size: Vector2.all(tileSize),
+      priority: 0,
+      sprite: spriteWater,
+    );
+    add(water3);
+
+    add(
+      QuadTreeDebugComponent(
+        collisionDetection as QuadTreeCollisionDetection,
+      ),
+    );
+    add(LayerComponent(staticLayer));
+    add(FpsTextComponent());
+    camera.zoom = 1;
+  }
+
+  final elapsedMicroseconds = <double>[];
+
+  late Player player;
+  final staticLayer = StaticLayer();
+  bool firstRender = true;
+  static const stepSize = 1.0;
+
+  @override
+  KeyEventResult onKeyEvent(
+    RawKeyEvent event,
+    Set<LogicalKeyboardKey> keysPressed,
+  ) {
+    Vector2? displacement;
+    var fireBullet = false;
+    for (final key in keysPressed) {
+      if (key == LogicalKeyboardKey.keyW && player.canMoveTop) {
+        displacement = Vector2(0, -stepSize);
+        player.position = player.position.translate(0, -stepSize);
+      }
+      if (key == LogicalKeyboardKey.keyA && player.canMoveLeft) {
+        displacement = Vector2(-stepSize, 0);
+        player.position = player.position.translate(-stepSize, 0);
+      }
+      if (key == LogicalKeyboardKey.keyS && player.canMoveBottom) {
+        displacement = Vector2(0, stepSize);
+        player.position = player.position.translate(0, stepSize);
+      }
+      if (key == LogicalKeyboardKey.keyD && player.canMoveRight) {
+        displacement = Vector2(stepSize, 0);
+        player.position = player.position.translate(stepSize, 0);
+      }
+      if (key == LogicalKeyboardKey.space) {
+        fireBullet = true;
+      }
+      if (key == LogicalKeyboardKey.keyO) {
+        (collisionDetection as QuadTreeCollisionDetection)
+            .quadBroadphase
+            .tree
+            .optimize();
+      }
+    }
+    if (fireBullet && displacement != null) {
+      final bullet =
+          Bullet(position: player.position, displacement: displacement * 50);
+      add(bullet);
```

Comment:
This will lead to that many bullets per tick are added, which might make faster devices able to add more bullets than slow devices. We might oversee that in this example though if the code becomes a lot longer by adding it in `update` instead.

Reply:
This is exactly what I intended to do: to create as much bullets as possible to force collision system to split screen into additional quadrants... and then you can press "O" and to see how these sections disappears: 
![image](https://user-images.githubusercontent.com/1258285/189505358-5c6e815f-e18a-412e-9120-fe527e99333f.png)